src/main/java/gov/usgs/wma/statistics/control/StatsService.java

```diff
@@ -66,86 +77,131 @@ public JsonData calculate(
+			}
+			builder.includeIntermediateValues(INCLUDE_MEDIANS.equals(medians));
+			// parse custom percentiles
+			if ( ! StatsService_PERCENTILES_DEFAULT.equals(percentiles) ) {
+				builder.percentiles(percentiles.split(","));
+				builder.buildPercentiles();
+			}
+
 			// A random identifier for the service unless we parameterize the date set ID.
 			Specifier spec = new Specifier();
-			
-			JsonData json = new WaterLevelStatistics(builder).calculate(spec, samples);
-			
-			LOGGER.trace("exited: good");
+
+			JsonData json;
+			// if there are parameter issues, then do not process statistics
+			if ( builder.hasErrors() ) {
+				json = builder.build();
+			} else {
+				json = new WaterLevelStatistics(env, builder).calculate(spec, samples);
+			}
+			LOGGER.trace("exited");
 			return json;
 		} catch (Exception e) {
-			LOGGER.trace("exited: b");
+			LOGGER.error("exited: ", e);
 			return null;
 		}
 	}
 	
 
 	// TODO should this be here, in a csv util class/lib, or someplace else.
-	public List<WLSample> parseData(String data){
+	public List<WLSample> parseData(String data, JsonDataBuilder builder){
 		String[] rows = data.split("\r?\n");
-		return parseData(rows);
+		return parseData(rows, builder);
 	}
-	public List<WLSample> parseData(String[] data) {
+	public List<WLSample> parseData(String[] data, JsonDataBuilder builder) {
 		
 		List<WLSample> samples = new ArrayList<>(data.length);
 		
-		for (String row : data) {
-			row = row.trim();
-			if ( 0 == row.length() || row.charAt(0) == '#') {
-				continue; // skip empty and comment rows
-			}
-			
-			String[] cols = row.split(",");
-			if (cols.length < 2 || cols.length > 3) {
-				throw new RuntimeException("All rows must have two values and optional provisional code: date,value,P. ");
-			}
-			
-			String time = cols[0].trim();
-			if ( time.length() < 10 ) {
-				throw new RuntimeException("The date must be valid (yyyy-mm-dd). " + time);
-			}
-			
+		String msg = "";
+		
+		for (int r=0; r<data.length; r++) {
+			String row = "";
 			try {
-				BigDecimal value = new BigDecimal(cols[1].trim());
+				if (isNotBlank(msg)) {
+					LOGGER.trace(msg);
+					msg = "";
+				}
+				row = data[r].trim();
+				if ( 0 == row.length() || row.charAt(0) == '#') {
+					continue; // skip empty and comment rows
+				}
+				
+				String[] cols = row.split(",");
+				if (cols.length < 2 || cols.length > 3) {
+					msg = env.getError(ENV_INVALID_ROW_COLS, r, row);
+					builder.error(msg);
+					continue;
+				}
+				
+				String time = cols[0].trim();
+				String val  = cols[1].trim();
 				
+				BigDecimal value = new BigDecimal(val);
 				WLSample sample = new WLSample(time, value, "ft", value, "", true, "", value);
 				
 				if (cols.length == 3) {
-					sample.setProvsional(true);
+					// Trim to remove whitespace and substring to convert full words like Provisional
+					String aging = cols[2].trim().substring(0, 1);
+					
+					if (Value.PROVISIONAL_CODE.equalsIgnoreCase(aging)) {
```

Comment:
Could split lines 188-195 into its own method to reduce nested if logic.

Reply:
yep